Add warnings when characters above 0xFF enter into Email::Simple

pali opened the pull request:
Whole Email::Simple works with bytes, not with Unicode strings. Therefore
add safe warnings about unexpected characters which cannot be represented
as bytes and which would cause serious problems.

For now only throw warnings, in future it can be fatal error.

rjbs commented:
I'm worried about new fatal things breaking long-"working" code, so I'll do a trial release of this soon.

rjbs commented:
> So lets first make it as a warning?

Sure.

pali committed:
Add warnings when characters above 0xFF enter into Email::Simple

Whole Email::Simple works with bytes, not with Unicode strings. Therefore
add safe warnings about unexpected characters which cannot be represented
as bytes and which would cause serious problems.

For now only throw warnings, in future it can be fatal error.

pali commented:
Done

pali commented:
@rjbs ping

jazzkutya commented:
RFC explicitly states that only 7bit ASCIIi is allowed so imho it should warn for anything above 0x7f or optionally it could be relaxed to anything above 0xff.

rjbs commented:
Present in trial release 2.217.
